src/elements/stream_element.ts

```diff
@@ -166,22 +166,34 @@ export class StreamElement extends HTMLElement {
   }
 
   private get targetElementsById() {
-    const element = this.ownerDocument?.getElementById(this.target!)
-
-    if (element !== null) {
-      return [element]
-    } else {
-      return []
+    const elementInDom = this.ownerDocument?.getElementById(this.target!)
+    
+    let foundElements : HTMLElement[] = [];
+    
+    if (elementInDom !== null) {
+      foundElements = [elementInDom]
     }
+
+    this.ownerDocument?.querySelectorAll("template").forEach((template) => {
```

Comment:
Since [querySelectorAll](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll) returns a `NodeList`, what do you think about using a `for` loop and `Array.push` here:

```js
const elements = []
const element = this.ownerDocument?.getElementById(this.target!)

if (element !== null ) {
  elements.push(element)
}

for (const template of this.ownerDocument?.querySelectorAll("template")) {
  const elementInTemplate = template.content.getElementById(this.target!)

  if (elementInTemplate) {
    elements.push(elementInTemplate)
  }
}

return elements
```

Reply:
Makes total sense. Thanks 👍 .